thoth-api/src/model/work/mod.rs

```diff
@@ -336,30 +336,46 @@ impl WorkStatus {
     fn is_withdrawn_out_of_print(&self) -> bool {
         matches!(self, WorkStatus::OutOfPrint | WorkStatus::WithdrawnFromSale)
     }
-
-    fn is_not_withdrawn_out_of_print(&self) -> bool {
-        !self.is_withdrawn_out_of_print()
-    }
 }
 
 pub trait WorkProperties {
     fn work_status(&self) -> &WorkStatus;
+    fn publication_date(&self) -> &Option<NaiveDate>;
     fn withdrawn_date(&self) -> &Option<NaiveDate>;
 
-    fn is_not_withdrawn_out_of_print(&self) -> bool {
-        self.work_status().is_not_withdrawn_out_of_print()
+    fn is_withdrawn_out_of_print(&self) -> bool {
+        self.work_status().is_withdrawn_out_of_print()
     }
 
     fn has_withdrawn_date(&self) -> bool {
         self.withdrawn_date().is_some()
     }
 
     fn withdrawn_date_error(&self) -> ThothResult<()> {
-        if self.is_not_withdrawn_out_of_print() && self.has_withdrawn_date() {
+        if !self.is_withdrawn_out_of_print() && self.has_withdrawn_date() {
             return Err(ThothError::WithdrawnDateError);
         }
         Ok(())
     }
+
+    fn no_withdrawn_date_error(&self) -> ThothResult<()> {
+        if self.is_withdrawn_out_of_print() && !self.has_withdrawn_date() {
+            return Err(ThothError::NoWithdrawnDateError);
+        }
+        Ok(())
+    }
+
+    fn withdrawn_date_before_publication_date_error(&self) -> ThothResult<()> {
+        match (self.withdrawn_date(), self.publication_date()) {
```

Comment:
You can access the values directly (and use them with `println`) instead of deconstructing them with `match`, e.g.

```
if let (Some(withdrawn_date), Some(publication_date)) = (self.withdrawn_date(), self.publication_date()) {
        if withdrawn_date < publication_date {
            println!("withdrawn_date: {:?}", withdrawn_date);
            println!("publication_date: {:?}", publication_date);
            return Err(ThothError::WithdrawnDateBeforePublicationDateError);
        }
    }
    ```